app/controllers/inventories_controller.rb

```diff
@@ -6,20 +6,48 @@
 
 class InventoriesController < ApplicationController
   before_action :require_can_view_projects!
-  before_action :require_can_edit_projects!, only: [:edit, :update]
-  before_action :set_inventory, only: [:show, :update, :edit]
-  before_action :set_project, only: [:edit, :update]
+  before_action :require_can_edit_projects!, only: [:new, :create, :edit, :update]
+  before_action :set_inventory, only: [:show, :update, :edit, :destroy]
+  before_action :set_project, only: [:edit, :update, :destroy]
+  before_action :set_project_from_params, only: [:new, :create]
 
   include ArelHelper
 
+  def new
+    @inventory = @project.inventories.build
+    @fields = new_form_fields
+    render layout: 'ajax_modal_rails/content'
+  end
+
+  def create
+    inventory = @project.inventories.new(
+      **manual_inventory_params,
+      manual_entry: true,
+      ProjectID: @project.ProjectID,
+      DateCreated: Time.now,
+      DateUpdated: Time.now,
+    )
+    inventory.save
+    redirect_to project_path(@project)
+  end
+
   def edit
+    return unless @inventory.manual_entry?
+
+    @fields = new_form_fields
+    render :edit_manual if @inventory.manual_entry?
   end
 
   def update
     @inventory.update(inventory_params)
     respond_with(@inventory, location: project_path(@project))
   end
 
+  def destroy
+    @inventory.destroy if @inventory.manual_entry?
+    redirect_to project_path(@project)
```

Comment:
You can use respond_with here as well.  It will give a nice flash message.

Reply:
✅